### ISSUE TITLE
peer_control: no way to close connection to a peer in state GOSSIPING

### ISSUE THREAD

**Issue by icota** [ElementsProject/lightning#1184]
`peer_from_json` doesn't iterate over results of `fromwire_gossip_getpeers_reply`. This is however done in `gossipd_getpeers_complete` when we call `listpeers` over RPC.

### Issue and Steps to Reproduce
```
~/Code/lightning$ ./cli/lightning-cli connect 02fa77e0f4ca666f7d158c4bb6675d1436e339903a9feeeaacbd6e55021b98e7ee 159.203.125.125
{ "id" : "02fa77e0f4ca666f7d158c4bb6675d1436e339903a9feeeaacbd6e55021b98e7ee" }
```

```
~/Code/lightning$ ./cli/lightning-cli listpeers
{ "peers" : 
        [ 
                { "state" : "GOSSIPING", "id" : "02fa77e0f4ca666f7d158c4bb6675d1436e339903a9feeeaacbd6e55021b98e7ee", "netaddr" : 
                        [ "159.203.125.125:9735" ], "connected" : true, "owner" : "lightning_gossipd" } ] }
```

```
~/Code/lightning$ ./cli/lightning-cli close 02fa77e0f4ca666f7d158c4bb6675d1436e339903a9feeeaacbd6e55021b98e7ee
{ "code" : -1, "message" : "Could not find peer with that id" }
```

**Comment by jBarz** [comment]
FYI, https://github.com/ElementsProject/lightning/pull/1181 implements a disconnect command

**Comment by ZmnSCPxj** [comment]
Closed in #1181
